compiler/test/fail_compilation/test16193.d

```diff
@@ -2,8 +2,8 @@
 REQUIRED_ARGS: -preview=dip1000
 TEST_OUTPUT:
 ---
-fail_compilation/test16193.d(38): Error: function `test16193.abc` is `@nogc` yet allocates closures with the GC
-fail_compilation/test16193.d(40):        test16193.abc.__foreachbody2 closes over variable x at fail_compilation/test16193.d(39)
+fail_compilation/test16193.d(38): Error: function `test16193.abc` is `@nogc` yet allocates closure for `abc()` with the GC
```

Comment:
This seem to fix two issues, one for the `@nogc` being ignored the other about the error message improvements.

Reply:
Sure, but it's merely a message improvement so it can be done separately.